eppoclient/lruassignmentlogger.go

```diff
@@ -22,20 +24,20 @@ type cacheValue struct {
 	variation  string
 }
 
-func NewLruAssignmentLogger(logger IAssignmentLogger, cacheSize int) IAssignmentLogger {
+func NewLruAssignmentLogger(logger IAssignmentLogger, cacheSize int) (IAssignmentLogger, error) {
 	cache, err := lru.New2Q[cacheKey, cacheValue](cacheSize)
 	if err != nil {
 		// err is only returned if `cacheSize` is invalid
 		// (e.g., <0) which should normally never happen.
-		panic(err)
+		return nil, fmt.Errorf("failed to create LRU cache: %w", err)
 	}
 	return &LruAssignmentLogger{
 		cache: cache,
 		inner: logger,
-	}
+	}, nil
 }
 
-func (self *LruAssignmentLogger) LogAssignment(event AssignmentEvent) {
```

Comment:
the linter reported that `self` or `this` was not desirable and that the variable should be more descriptive; I have used a shorthand for the struct previously.

Reply:
Coming from languages that enforce `self`, that's pretty much unfortunate. I don't think `lal` is any more descriptive than `self` here.

Reply:
how about `logger` or `assignmentLogger`?

Reply:
It is common to name the `method received` with one character, the lowercase version of the first character in the implementing struct.

https://go.dev/wiki/CodeReviewComments#receiver-names

```
Receiver Names

The name of a method’s receiver should be a reflection of its identity; often a one or two letter abbreviation of its type suffices (such as “c” or “cl” for “Client”). Don’t use generic names such as “me”, “this” or “self”, identifiers typical of object-oriented languages that gives the method a special meaning. In Go, the receiver of a method is just another parameter and therefore, should be named accordingly. The name need not be as descriptive as that of a method argument, as its role is obvious and serves no documentary purpose. It can be very short as it will appear on almost every line of every method of the type; familiarity admits brevity. Be consistent, too: if you call the receiver “c” in one method, don’t call it “cl” in another.
```